### PR TITLE
implement deploy logs view for helm managed mode

### PR DESCRIPTION
**Pull request by stefanrepl**
#### What this PR does / why we need it:
This PR implements deploy logs API response for KOTS when running in helm managed mode
#### Which issue(s) this PR fixes:

Fixes #
https://app.shortcut.com/replicated/story/54248/404-returned-when-trying-to-view-deploy-logs
#### Does this PR introduce a user-facing change?

```release-note
     Show deploy logs of helm charts when running in [Helm managed mode (Alpha)](/vendor/helm-install).
```

#### Does this PR require documentation?

Loom: https://www.loom.com/share/9ea4867e71fc4087b572d9f7d53b7c20

### CODE REVIEW

**Comment by divolgin** [comment]
Presumably this is received in the watch loop with secret info.  Is there a revision number?  The output should match the correct revision, but this will be shown all revisions of the app.  The API handler receives sequence number.  It should be used the retrieve matching output.

File: pkg/handlers/downstream.go [hunk 1]
```diff
@@ -26,24 +29,40 @@ func (h *Handler) GetDownstreamOutput(w http.ResponseWriter, r *http.Request) {
...
 
-	a, err := store.GetStore().GetAppFromSlug(appSlug)
-	if err != nil {
-		logger.Error(err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
+	if util.IsHelmManaged() {
+		app := helm.GetHelmApp(appSlug)
+		releaseSecret, err := helm.GetChartSecret(app.Release.Name, app.Release.Namespace, mux.Vars(r)["sequence"])
```
Comment (divolgin) [inline pkg/handlers/downstream.go:41]: if app == nil, return NotFound

File: pkg/helm/charts.go [hunk 2]
```diff
@@ -59,6 +59,40 @@ func (v InstalledReleases) Less(i, j int) bool {
...
+		"version": version,
+	}
+	listOpts := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(selectorLabels).String(),
+	}
+
+	secrets, err := clientSet.CoreV1().Secrets(namespace).List(context.TODO(), listOpts)
+	if err != nil {
+		if kuberneteserrors.IsNotFound(err) {
+			return &helmrelease.Release{}, nil
```
Comment (divolgin) [inline pkg/helm/charts.go:80]: `return nil, nil` or just remove this check.  the caller should check that the app is not found.

The problem is that `Release` contains pointers that will be nil here and will cause panics.  `Info` is a pointer and `releaseSecret.Info.Description` will panic.

File: pkg/handlers/downstream.go (hunk 1 above)
Comment (stefanrepl) [reply pkg/handlers/downstream.go:45]: done

File: pkg/helm/charts.go (hunk 2 above)
Comment (stefanrepl) [reply pkg/helm/charts.go]: done, returning nil, nil

File: pkg/handlers/downstream.go [hunk 3]
```diff
@@ -26,24 +29,44 @@ func (h *Handler) GetDownstreamOutput(w http.ResponseWriter, r *http.Request) {
...
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		releaseSecret, err := helm.GetChartSecret(app.Release.Name, app.Release.Namespace, mux.Vars(r)["sequence"])
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		if releaseSecret.Info.Status != "failed" {
```
Comment (divolgin) [inline pkg/handlers/downstream.go:51]: this needs to check for releaseSecret  != nil now

Comment (stefanrepl) [reply pkg/handlers/downstream.go]: ah of course, thanks

File: web/src/components/snapshots/SnapshotDetails.jsx [hunk 4]
```diff
@@ -86,23 +86,23 @@ class SnapshotDetails extends Component {
       tooltip: {
         custom: function ({ series, seriesIndex, dataPointIndex, w }) {
           return (
-            '<div class="arrow_box">' +
-            '<p class="u-textColor--primary u-fontSize--normal u-fontWeight--medium">' +
+            "<div class=\"arrow_box\">" +
```
Comment (stefanrepl) [inline web/src/components/snapshots/SnapshotDetails.jsx:89]: my IDE's JS linter did these quoting changes, but seems to make sense

**Review by divolgin — APPROVED** [review]

**Pull request merged**